File: ApiUsuarios/src/main/java/org/example/apiusuarios/dto/ComidaDTO.java
package org.example.apiusuarios.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.example.apiusuarios.model.Comida;
import org.example.apiusuarios.model.TipoComida;
import org.example.apiusuarios.model.UnidadComida;
import java.time.LocalDate;

@Data
@NoArgsConstructor
public class ComidaDTO {

    private Integer comidaId;
    private String nombre;
    private Integer calorias;
    private Double grasasSaturadas;
    private Double proteinas;
    private Double carbohidratos;
    private TipoComida tipoComida;
    private Double cantidad;
    private UnidadComida unidad;
    private LocalDate fecha;
    private Integer usuarioId;

    public ComidaDTO(Comida c) {
        this.comidaId = c.getComidaId();
        this.nombre = c.getNombre();
        this.calorias = c.getCalorias();
        this.grasasSaturadas = c.getGrasasSaturadas();
        this.proteinas = c.getProteinas();
        this.carbohidratos = c.getCarbohidratos();
        this.tipoComida = c.getTipoComida();
        this.cantidad = c.getCantidad();
        this.unidad = c.getUnidad();
        this.fecha = c.getFecha();
        this.usuarioId = c.getUsuario() != null ? c.getUsuario().getUsuarioId() : null;
    }
}

File: FitWinKMP/composeApp/src/commonMain/kotlin/org/example/fitwinkmp/features/food/data/dto/ComidaDTO.kt
package org.example.fitwinkmp.features.food.data.dto

import kotlinx.serialization.Serializable

@Serializable
data class ComidaDTO(
    val comidaId: Int? = null,
    val usuarioId: Int,
    val nombre: String,
    val calorias: Double,
    val proteinas: Double,
    val carbohidratos: Double,
    val grasasSaturadas: Double,
    val tipoComida: String, // DESAYUNO, ALMUERZO, CENA, SNACK
    val cantidad: Double,
    val unidad: String, // GRAMOS, ML, UNIDAD
    val fecha: String // YYYY-MM-DD
)
